src/getMachineId.py

```diff
@@ -0,0 +1,44 @@
+import boto3
+from decEncoder import *
+from boto3.dynamodb.conditions import Key, Attr
+
+"""Dynamo resource"""
+dynamodb = boto3.resource('dynamodb', region_name = 'us-east-2')
+machineTable = dynamodb.Table('Machines')
+
+"""Lambda handler function for /buildings/{buildingId}/machines/{machineId} API call
+   Returns machine with the machineId specified in the path
+   or 'Building not found' error if buildingId doesn't match buildingId of machine
+   or 'Machine not found' error if machineId not found by get_item search
+"""
+def getMachineId(event, context):
+    """If machineId specified, assign it to variable,
+       use get_item to find it in machine table
+       if it finds a machine, verify it's in building specified in path
+	   if it is return machine info in JSON format
+    """
+    buildingId = None
+    machineId = None
+    if event.get('pathParameters'):
```

Comment:
I don't particularly love the idea of having an if statement without an else in this case. API gateway should, at least in theory, ensure we always get a path parameter in this case. You can probably get rid of this. Alternatively, you can return a 400 (https://tools.ietf.org/html/rfc7231#section-6.5.1)